In [1]:
!pip install datasets
!pip install transformers
!pip install evaluate
import evaluate

from datasets import load_dataset
imdbDataset = load_dataset("imdb")
yelpDataset = load_dataset("yelp_review_full")

from transformers import pipeline

bert_pipelineTrun = pipeline("sentiment-analysis", model="nlptown/bert-base-multilingual-uncased-sentiment", truncation=True)
distilbert_pipelineTrun = pipeline("sentiment-analysis", model="distilbert-base-uncased-finetuned-sst-2-english", truncation=True)

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 480.6/480.6 kB 19.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 116.3/116.3 kB 8.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 179.3/179.3 kB 12.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 143.5/143.5 kB 13.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 194.8/194.8 kB 10.9 MB/s eta 0:00:00
  Attempting uninstall: fsspec
    Found existing installation: fsspec 2024.10.0
    Uninstalling fsspec-2024.10.0:
      Successfully uninstalled fsspec-2024.10.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
gcsfs 2024.10.0 requires fsspec==2024.10.0, but you have fsspec 2024.9.0 which is incompatible.
torch 2.5.1+cu124 requires nvidia-cublas-cu12==12.4.5.8; platform_system == "Linux" and platform_machine == "x86_64", but you have nvidia-cublas-cu12 12.5.3.2 wh

/usr/local/lib/python3.11/dist-packages/huggingface_hub/utils/_auth.py:94: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


README.md:   0%|          | 0.00/7.81k [00:00<?, ?B/s]

train-00000-of-00001.parquet:   0%|          | 0.00/21.0M [00:00<?, ?B/s]

test-00000-of-00001.parquet:   0%|          | 0.00/20.5M [00:00<?, ?B/s]

unsupervised-00000-of-00001.parquet:   0%|          | 0.00/42.0M [00:00<?, ?B/s]

Generating train split:   0%|          | 0/25000 [00:00<?, ? examples/s]

Generating test split:   0%|          | 0/25000 [00:00<?, ? examples/s]

Generating unsupervised split:   0%|          | 0/50000 [00:00<?, ? examples/s]

README.md:   0%|          | 0.00/6.72k [00:00<?, ?B/s]

train-00000-of-00001.parquet:   0%|          | 0.00/299M [00:00<?, ?B/s]

test-00000-of-00001.parquet:   0%|          | 0.00/23.5M [00:00<?, ?B/s]

Generating train split:   0%|          | 0/650000 [00:00<?, ? examples/s]

Generating test split:   0%|          | 0/50000 [00:00<?, ? examples/s]

config.json:   0%|          | 0.00/953 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/669M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/39.0 [00:00<?, ?B/s]

vocab.txt:   0%|          | 0.00/872k [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/112 [00:00<?, ?B/s]

Device set to use cuda:0


config.json:   0%|          | 0.00/629 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/268M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/48.0 [00:00<?, ?B/s]

vocab.txt:   0%|          | 0.00/232k [00:00<?, ?B/s]

Device set to use cuda:0


In [2]:
imdbReviews = imdbDataset["test"]["text"]
imdbLabels = imdbDataset["test"]["label"]

yelpReviews = yelpDataset["test"]["text"]
yelpTrueLabels = yelpDataset["test"]["label"]

bertImdbResults = bert_pipelineTrun(imdbReviews)
distilbertImdbResults = distilbert_pipelineTrun(imdbReviews)

bertYelpResults = bert_pipelineTrun(yelpReviews)
distilbertYelpResults = distilbert_pipelineTrun(yelpReviews)

In [3]:
yelpLabelMapping = {"0": 0, "1": 0, "2": 1, "3": 1, "4": 1}
yelpTrueLabels = [yelpLabelMapping[str(label)] for label in yelpTrueLabels]

bertImdbLabels = [result["label"] for result in bertImdbResults]
distilbertImdbLabels = [result["label"] for result in distilbertImdbResults]

bertImdbLabelMapping = {"1 star": 0, "2 stars": 0, "3 stars": 1, "4 stars": 1, "5 stars": 1}
distilbertImdbLabelMapping = {"NEGATIVE": 0, "POSITIVE": 1}
bertImdbLabels = [bertImdbLabelMapping[label] for label in bertImdbLabels]
distilbertImdbLabels = [distilbertImdbLabelMapping[label] for label in distilbertImdbLabels]

bertYelpLabels = [result["label"] for result in bertYelpResults]
distilbertYelpLabels = [result["label"] for result in distilbertYelpResults]

bertYelpLabelMapping = {"1 star": 0, "2 stars": 0, "3 stars": 1, "4 stars": 1, "5 stars": 1}
distilbertYelpLabelMapping = {"NEGATIVE": 0, "POSITIVE": 1}
distilbertYelpLabels = [distilbertYelpLabelMapping[label] for label in distilbertYelpLabels]
bertYelpLabels = [bertYelpLabelMapping[label] for label in bertYelpLabels]

In [4]:
accuracy_metric = evaluate.load("accuracy")
f1_metric = evaluate.load("f1")

bertAccuracy = accuracy_metric.compute(predictions=bertImdbLabels, references=imdbLabels)
bertF1 = f1_metric.compute(predictions=bertImdbLabels, references=imdbLabels)

print("IMDB Reviews:")
print("BERT Accuracy (Truncated):", bertAccuracy["accuracy"])
print("BERT F1 Score (Truncated):", bertF1["f1"])

distilbertAccuracy = accuracy_metric.compute(predictions=distilbertImdbLabels, references=imdbLabels)
distilbertF1 = f1_metric.compute(predictions=distilbertImdbLabels, references=imdbLabels)

print("DistilBERT Accuracy (Truncated):", distilbertAccuracy["accuracy"])
print("DistilBERT F1 Score (Truncated):", distilbertF1["f1"])

IMDB Reviews:
BERT Accuracy (Truncated): 0.8556
BERT F1 Score (Truncated): 0.8650063570413582
DistilBERT Accuracy (Truncated): 0.89072
DistilBERT F1 Score (Truncated): 0.8874794069192751


In [5]:
bertYelpAccuracy = accuracy_metric.compute(predictions=bertYelpLabels, references=yelpTrueLabels)
bertYelpF1 = f1_metric.compute(predictions=bertYelpLabels, references=yelpTrueLabels, average="weighted")

print("Yelp Reviews:")
print("BERT Accuracy (Truncated):", bertYelpAccuracy["accuracy"])
print("BERT F1 Score (Truncated):", bertYelpF1["f1"])

distilbertYelpAccuracy = accuracy_metric.compute(predictions=distilbertYelpLabels, references=yelpTrueLabels)
distilbertYelpF1 = f1_metric.compute(predictions=distilbertYelpLabels, references=yelpTrueLabels, average="weighted")

print("DistilBERT Accuracy (Truncated):", distilbertYelpAccuracy["accuracy"])
print("DistilBERT F1 Score (Truncated):", distilbertYelpF1["f1"])

Yelp Reviews:
BERT Accuracy (Truncated): 0.86278
BERT F1 Score (Truncated): 0.8623747864713464
DistilBERT Accuracy (Truncated): 0.82458
DistilBERT F1 Score (Truncated): 0.8263340259137584


---
# Truncated Method:
Models:
- "nlptown/bert-base-multilingual-uncased-sentiment"
- "distilbert-base-uncased-finetuned-sst-2-english"

Datasets:
- IMDB Reviews
- Yelp Reviews

##Results

    IMDB Reviews:

    BERT Accuracy (Truncated): 0.8556
    BERT F1 Score (Truncated): 0.8650063570413582
    DistilBERT Accuracy (Truncated): 0.89072
    DistilBERT F1 Score (Truncated): 0.8874794069192751

‎

    Yelp Reviews:

    Yelp -> BERT Accuracy (Truncated): 0.86278
    Yelp -> BERT F1 Score (Truncated): 0.8623747864713464
    Yelp -> DistilBERT Accuracy (Truncated): 0.82458
    Yelp -> DistilBERT F1 Score (Truncated): 0.8263340259137584
---

# Chunking Method:

Models:
- "nlptown/bert-base-multilingual-uncased-sentiment"
- "distilbert-base-uncased-finetuned-sst-2-english"

Datasets:
- IMDB Reviews
- Yelp Reviews

---




